File: cont/base/springcontent/shaders/GLSL/Particles/ExploSpikeParticleGenerator.lua
return {
	MainCode =
[[
	float a = max(0.0f, alpha - alphaDecay * frameInfo.y);
	color *= a;
	
	vec3 dif1 = normalize(pos - camPos);
	vec3 dir2 = normalize(cross(dif1, dir));
	
	vec3 l = (dir * length) + (lengthGrowth * frameInfo.y);
	vec3 w = (dir2 * width);
	
	vec3 drawPos = pos + speed * frameInfo.y;

	AddEffectsQuad(
		drawOrder,
		vec3(1.0),
		drawPos - l - w,
		drawPos + l - w,
		drawPos + l + w,
		drawPos - l + w,
		texCoord,
		color
	);
]]
}

File: rts/Rendering/Env/Particles/Generators/ExploSpikeParticleGenerator.cpp
#include "ExploSpikeParticleGenerator.h"

CR_BIND(ExploSpikeParticleData, )
CR_REG_METADATA(ExploSpikeParticleData,
(
	CR_MEMBER(pos),
	CR_MEMBER(alpha),
	CR_MEMBER(speed),
	CR_MEMBER(alphaDecay),
	CR_MEMBER(dir),
	CR_MEMBER(color),
	CR_MEMBER(length),
	CR_MEMBER(lengthGrowth),
	CR_MEMBER(width),
	CR_MEMBER(drawOrder),
	CR_MEMBER(texCoord)
))